### DISCUSSION TITLE
Trouble Access Config in Flow

### DISCUSSION THREAD

**Discussion by pdxwebdev** [JanssenProject/jans#8000] [General]
https://github.com/pdxwebdev/centeridentityagama/blob/agama-lab-branch/projects/secretlocationauth

I'm trying to access the conf variable in various places in the flow and it appears to be an empty object {}.

In the template, it attempts to list sign in options using the keys from the config variable but the list appears to be blank, no sign in options listed.

This is the config value for the start of the flow:
```
{
  "com.centeridentity.oidc": {
    "CenterIdentity": {
      "host": "https://centeridentity.com",
      "dcr": {
        "enabled": true,
        "useCachedClient": true
      },
      "oauth": {
        "scopes": [
          "openid"
        ]
      },
      "provision": {
        "uidPrefix": "centeridentity-",
        "attribute": "sub"
      }
    },
    "Github": {
      "oauth": {
        "authzEndpoint": "https://github.com/login/oauth/authorize",
        "tokenEndpoint": "https://github.com/login/oauth/access_token",
        "userInfoEndpoint": "https://api.github.com/user",
        "clientId": "mangled",
        "clientSecret": "twisted",
        "scopes": [
          "user"
        ]
      },
      "provision": {
        "uidPrefix": "github-",
        "attribute": "login"
      }
    }
  }
}
```
Given the config, the following template excerpt, it should output two options, one for CenterIdentity and one for GitHub. In this case, providers = config (the json above).
```
[#list providers?keys as id]
<div class="text-sm flex justify-center items-center">
	<button type="submit" name="selection" value="${id}" class="w-80 py-2 px-6 rounded text-white btn bg-green-500 hover:bg-green-600">${id}</button>
</div>
[/#list]
```

But the list is blank as seen in the screenshot. I'm not sure what's going wrong.
![Screenshot from 2024-03-07 17-24-01](https://github.com/JanssenProject/jans/assets/490176/c859853f-9f72-4bc6-8c66-b91dd9f84663)

**Comment by nynymike** [comment]
`https://centeridentity.com/.well-known/openid-configuration` returns: 

```json
{
"issuer": "https://centeridentity.com", 
"authorization_endpoint": "https://centeridentity.com/oauth2/openid-login", 
"token_endpoint": "https://centeridentity.com/oauth2/openid-token", 
"userinfo_endpoint": "https://centeridentity.com/oauth2/openid-userinfo", 
"jwks_uri": "https://centeridentity.com/.well-known/jwks.json"
}
```

You can't do DCR if there is no `registration_endpoint`...

```json
"dcr" :  {
        "enabled": true
```

**Comment by nynymike** [comment]
BTW, why do you even need the `RRF`, why don't you just jump to the `Trigger`... you know which IDP you want to demo your authn service.

**Reply by pdxwebdev** [reply]
My project is basically forked from the agama-openid repo and made as little modifications as possible just to get a baseline in case my modifications had something to do with it. So that's why some of the other stuff like dcr still remains.

From my observation, is has not called the /openid-configuration endpoint yet.

**Reply by nynymike** [reply]
Perhaps it's not showing you the list of OP's because DCR failed. Do you see any messages in `/opt/jans/jetty/jans-auth/logs`
-  Mike

**Comment by pdxwebdev** [comment]
Ahhh, got it! 

For posterity:
The configuration had to be imported in the TUI.

![Screenshot from 2024-03-08 10-09-33](https://github.com/JanssenProject/jans/assets/490176/e810dee2-dfec-494b-a2d7-adefe212c1d1)

**Reply by nynymike** [reply]
My video pretty clearly shows this: https://gluu.co/apow-4
